pkg/provisioner/ironic/ironic.go

```diff
@@ -1217,10 +1217,7 @@ func (p *ironicProvisioner) Prepare(data provisioner.PrepareData, unprepared boo
 			started = true
 		}
 		// Manual clean finished
-		result, err = p.changeNodeProvisionState(
-			ironicNode,
-			nodes.ProvisionStateOpts{Target: nodes.TargetProvide},
-		)
+		result, err = operationComplete()
```

Comment:
If we remove this change of the provision state, how will the ironic host make the manageable->cleaning->available transition?

Ref https://docs.openstack.org/ironic/latest/_images/states.svg

Reply:
Could someone please answer this question?

Reply:
As @Hellcatlk has already replied [here](https://github.com/metal3-io/baremetal-operator/pull/929#issuecomment-882267676). If we remove the changes here, then the auto cleaning will be executed in [`provision`](https://github.com/metal3-io/baremetal-operator/blob/493c56fc9b120ce4c58ac25ef8eb63d276e54528/pkg/provisioner/ironic/ironic.go#L1356-L1358), so the ironic host can still make the manageable->cleaning->available transition.